### PR TITLE
Ordering content by weight

### LINKED ISSUE #8702: Menu in reverse order
**This is a...** 

- [ ] Feature Request
- [x] Bug Report

**Problem:**
A menu is in reverse order. In the image below see Annotations. That list is in reverse order. Annotations should be at the bottom.

screen shot 2018-05-23 at 1 37 22 pm

### PR DESCRIPTION
**Pull request by cstoku**
Fix #8702

I set the weights according to the following procedure.

1. If weight is set, keep its weight
1. If not set, weight is set with reference to the order of document of version 1.9

The contents that weight can not be set in the above procedure are as follows.
I did not set weights for these contents.

- [Installing Addons](https://kubernetes.io/docs/concepts/cluster-administration/addons/)

I’m sorry for the big PR :bowing_man: 
Thank you!

### CODE REVIEW

**Comment by tengqm** [comment]
/lgtm

I didn't go through all the weight values assigned. LGTM because I think having weights assigned is already much better than the default ordering. We can always adjust the weights later if needed.

**Comment by steveperry-53** [comment]
@cstoku Thank you for this pull request. I'm sorry to say that we cannot use much of what you've done here. I have already added weights for Tasks and Tutorials.
Tasks: #8601, #8602, #8623, #8625, #8634, #8635, #8582
Tutorials: #8557

So you'll need to remove all files under Tasks and Tutorials from this pull request.

I'm not sure what the situation is for topics under Concepts.
Issue #8702 is a duplicate of Issue #8239, which covers the Concepts section.
And @chrisnegus has already signed up to do that work.

@chrisnegus, Have you started work on reordering the left nav for Concepts. If not, could you maybe coordinate with @cstoku, and use some of the work that has been done in this PR?

**Comment by cstoku** [comment]
It's okay. I'm rather sorry that I missed it. 🙇 

I will revert to the commit of contents of Tasks and Tutorials.

**Comment by chrisnegus** [comment]
It looks like @cstoku has made more progress than I have. I'd be fine with @cstoku taking over this work.

**Comment by zacharysarah** [comment]
@steveperry-53 It looks like the question of which approach to take is resolved. Can we remove the hold?

File: layouts/partials/tree.html [hunk 1]
```diff
@@ -1,25 +1,27 @@
-{{ $section := . }}
```
Comment (chenopis) [inline layouts/partials/tree.html]: @bep Can you take a quick look and make sure this looks good to you? Thx

**Comment by steveperry-53** [comment]
@cstoku, Can you explain your reason for changing layouts/partials/tree.html? At first glance, I don't see any need to change that file. Thanks.

File: layouts/partials/tree.html (hunk 1 above)
Comment (bep) [reply layouts/partials/tree.html]: This does not look good.

**Comment by bep** [comment]
One general comment from me, coming from a "content management perspective"; I understand that this PR comes from getting same/similar ordering as the old Jekyll version. If that requirement comes from a "this particular order is really important", that is fine -- but remember that the maintenance of these weight has a cost in itself (this gets even more visible if you get 20 translations and want to preserve the same order in all of them). If this was my project, I would probably use weight to bump the more important sections/pages to the top, and keep the rest in its default order. The sorting is stable, out of the box, which I think is the key here.

**Comment by cstoku** [comment]
@steveperry-53 
In the template before change, Page and Section are sorted separately.
Specifically, Sections are sorted after Page sorting.

In this PR, Page and Section are combined and sorted by weight.

A concrete example is in Concepts -> Extending Kubernetes

(Actually, I did not know that if the weight was set on the page, hugo would sort it considering weight. :sweat_smile:)

**Pull request merged**